### PR TITLE
Add Array errorhandling

### PR DESCRIPTION
**Pull request by DashieTM**
## Proposed Changes
This simply adds the functionality from lists to arrays by mirroring the functionality.
Alternative way would be to just always convert Arrays to and from lists.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Build and tests pass locally
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/FsToolkit.ErrorHandling/Array.fs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    let sequenceAsyncOptionM xs = traverseAsyncOptionM id xs
+
+#if !FABLE_COMPILER
+    let rec private traverseVOptionM' (state: voption<_>) (f: _ -> voption<_>) xs =
+        match xs with
+        | [||] ->
+            state
+            |> ValueOption.map Array.rev
+        | arr ->
+            let x = Array.head arr
```
Comment (bartelink) [inline src/FsToolkit.ErrorHandling/Array.fs:212]: isnt there an array.split or similar that can do this pair of ops one shot?

File: src/FsToolkit.ErrorHandling/Array.fs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            let xs = Array.skip 1 arr
+            let fR = f x
+
+            match state, fR with
+            | Ok ys, Ok y -> traverseValidationA' (Ok(Array.append [| y |] ys)) f xs
+            | Error errs1, Error errs2 ->
+                let errs =
+                    Array.concat [
```
Comment (bartelink) [inline src/FsToolkit.ErrorHandling/Array.fs:121]: Array.append? (or at least don't allocate a `list` via `[]`?

File: src/FsToolkit.ErrorHandling/Array.fs (hunk 1 above)
Comment (DashieTM) [reply src/FsToolkit.ErrorHandling/Array.fs:212]: Not in a one shot, it returns 2 arrays, which is not what we want here.

File: src/FsToolkit.ErrorHandling/Array.fs (hunk 2 above)
Comment (DashieTM) [reply src/FsToolkit.ErrorHandling/Array.fs]: Thanks!

File: src/FsToolkit.ErrorHandling/Array.fs (hunk 1 above)
Comment (bartelink) [reply src/FsToolkit.ErrorHandling/Array.fs:207]: true, not sure what I was thinking!

Comment (DashieTM) [reply src/FsToolkit.ErrorHandling/Array.fs:207]: All good, no worries :D

**Pull request merged**